server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java

```diff
@@ -37,7 +38,8 @@
  */
 public class RemoteClusterPortSettings {
 
-    public static final TransportVersion TRANSPORT_VERSION_REMOTE_CLUSTER_SECURITY = TransportVersion.V_8_8_0;
+    public static final TransportVersion TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY = TransportVersion.V_8_8_0;
+    public static final Version VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY = Version.V_8_8_0;
```

Comment:
For the constants I'm replacing, we are using `TransportVersion` in some places, and `Version` in others. We should consolidate this (if it's actually redundant), but I'd prefer to do that separately from this PR, so I'm introducing a new `Version` typed constant here.

Reply:
Separate PR sounds good since it touches quite a few files. I am pretty sure it should be replaced with transport version as well and `getMinNodeVersion` should be replaced with `clusterService.state().getMinTransportVersion()` similar to how it is done for [TokenService](https://github.com/elastic/elasticsearch/blob/97d1ca05958cba0b24cd55b2367ab569feaab226/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java#L1432).